drivers/adc/Kconfig.mcux

```diff
@@ -15,6 +15,12 @@ config ADC_MCUX_ADC16
 	help
 	  Enable the MCUX ADC16 driver.
 
+config ADC_MCUX_RTADC
```

Comment:
The convention for MCUXpresso shim drivers is: `<driver class>_MCUX_<HW IP name>`. So for this driver, it should be `ADC_MCUX_12B1MSPS_SAR`. See https://github.com/zephyrproject-rtos/hal_nxp/blob/4560b0a0106bbe47eded4d7af775998c4f6d05c0/mcux/drivers/imx/fsl_adc.c#L11-L14

Reply:
I saw 3 mcux adc drivers for another nxp controllers: ADC_MCUX_ADC12, ADC_MCUX_ADC16, ADC_MCUX_LTADC. Should I rename RT driver or leave naming like other nxp mcux drivers?

Reply:
Yes, please rename the RT driver.

Reply:
I dhange name to driver config. If I need to rename all entries, write plese.

Reply:
Yes, please

Reply:
Sorry, but I can't understand what should I rename? Should I rename driver file with this name or not? Because I see that zephyr files and driver condigs don't have names with 3 underlines

Reply:
Yes please rename this driver with this name, the additional underscore is ok. I admit that `12B1MSPS_SAR` is a little awkward, but I want to try to reflect the name of the HW IP block because sometimes 1) the same HW IP is used in multiple SoC families (e.g., LPUART in Kinetis and i.MX RT) or 2) there are multiple HW IP block options for the same driver class (e.g., ADC12 and ADC16 in Kinetis). Hence I don't want to use the SoC family name.

Reply:
Change driver name with 12B1MSPS_SAR label, remove SoC connection in comments.